packages/token-balance-cache/src/providers/erc1155.ts

```diff
@@ -0,0 +1,77 @@
+import Web3 from 'web3';
+import { BigNumber, providers } from 'ethers';
+import type { ERC1155 } from 'common-common/src/eth/types';
+import { ERC1155__factory } from 'common-common/src/eth/types';
+
+import type { IChainNode } from '../types';
+import { BalanceProvider } from '../types';
+import { BalanceType } from 'common-common/src/types';
+
+type EthBPOpts = {
+  tokenAddress?: string;
+  contractType?: string;
+  tokenId?: string;
+};
+
+export default class Erc1155BalanceProvider extends BalanceProvider<EthBPOpts> {
+  public name = 'erc1155';
+  // Added Token Id as String because it can be a BigNumber (uint256 on solidity)
+  public opts = {
+    contractType: 'string?',
+    tokenAddress: 'string?',
+    tokenId: 'string?',
+  };
+  public validBases = [BalanceType.Ethereum];
+
+  public getCacheKey(
+    node: IChainNode,
+    address: string,
+    opts: EthBPOpts
+  ): string {
+    return `${node.id}-${address}-${opts.tokenAddress}`;
+  }
+
+  public async getBalance(
+    node: IChainNode,
+    address: string,
+    opts: EthBPOpts
+  ): Promise<string> {
+    const url = node.private_url || node.url;
```

Comment:
A lot of this code is shared between balance providers. Can we DRY it?

Reply:
As discussed, this came up pre-implementation and I believe we said we were good with repeating some code in each of the provider files as a first pass, and then we would circle back to DRY the code.

Reply:
Dryed!